static/src/javascripts/lib/url.ts

```diff
@@ -0,0 +1,131 @@
+import memoize from 'lodash/memoize';
+import { hasPushStateSupport } from 'lib/detect';
+
+// TODO: typescript detect.js
+const supportsPushState = hasPushStateSupport() as boolean;
+
+/* commercial testing instrument */
+// Returns a map { <bidderName>: true } of bidders
+// according to the pbtest URL parameter
+
+const pbTestNameMap: () => Record<string, true | undefined> = memoize(
+	(): Record<string, undefined | true> =>
+		new URLSearchParams(window.location.search)
+			.getAll('pbtest')
+			.reduce<Record<string, undefined | true>>((acc, value) => {
+				acc[value] = true;
+				return acc;
+			}, {}),
+	(): string =>
+		// Same implicit parameter as the memoized function
+		window.location.search,
+);
+
+// returns "foo=bar&fizz=buzz" (eg. no ? symbol)
+const getCurrentQueryString = (): string =>
+	window.location.search.replace(/^\?/, '');
```

Comment:
We could extract this regex for performance:

```suggestion
const STARTING_QUESTION_MARK = /^\?/
const getCurrentQueryString = (): string =>
	window.location.search.replace(STARTING_QUESTION_MARK, '');
```